Pipelines/Templates/export-Solution.yml

```diff
@@ -48,13 +48,14 @@ steps:
   workingDirectory: $(Build.SourcesDirectory)\${{parameters.Repo}}
   displayName: 'Initialize Git'
 
-# If the BranchToCreate variable *does not* contain an empty string, then
+# If the BranchToCreate parameter *does not* contain an empty string, then
 # create a branch based on BranchToCreate variable value and checkout that branch.
-- script: |
-   git checkout -b ${{parameters.BranchToCreate}} ${{parameters.Branch}}
-  displayName: 'Create New Git Branch: ${{parameters.BranchToCreate}}'
-  workingDirectory: $(Build.SourcesDirectory)\${{parameters.Repo}}
-  condition: and(succeeded(), ne(variables.BranchToCreate, '')) # If an empty value is passed for the BranchToCreate variable, then skip this task
+- ${{ if ne(parameters.BranchToCreate, '')}}: # If an empty value is passed for the BranchToCreate parameter, then skip this task
+  - script: |
+      git checkout -b ${{parameters.BranchToCreate}} ${{parameters.Branch}}
+    displayName: 'Create New Git Branch: ${{parameters.BranchToCreate}}'
+    workingDirectory: $(Build.SourcesDirectory)\${{parameters.Repo}}
+    condition: succeeded()
```

Comment:
@FH-Inway I finally got around to testing this out locally. Changes look good, but I have one piece of feedback. I'd prefer for consistency that we use conditions rather than the if statements here and else where in the YAML. This is primarily a stylistic choice unless there is a specific requirements for the way it's been implemented. e.g. change to  ```condition: and(succeeded(), ne('${{parameters.BranchToCreate}}', ''))```

Reply:
@mikefactorial Thanks for testing. There is actually a specific requirement for implementing it this way: conditions in templates do not support parameters passed to the template :) see [Use a template parameter as part of a condition](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml#use-a-template-parameter-as-part-of-a-condition)

> "When you declare a parameter **in the same pipeline** that you have a condition, parameter expansion happens before conditions are considered. In this case, you can embed parameters inside conditions. [...]
> However, when you pass a parameter to a template, the parameter will not have a value when the condition gets evaluated. As a result, if you set the parameter value in both the template and the pipeline YAML files, the pipeline value from the template will get used in your condition."

Reply:
Hey @FH-Inway just getting back to this again. The article you linked to contradicts what you said. ```When you declare a parameter in the same pipeline that you have a condition, parameter expansion happens before conditions are considered. In this case, you can embed parameters inside conditions.``` Not sure I follow...

Reply:
Hey @mikefactorial , thanks for getting back to this. Probably just a misunderstanding, but in this case we have a yaml template and not a pipeline, which is why the parameter cannot be embeded inside conditions. If you want, we could have a quick call on this, probably easier to talk about it. You can ping me on Teams at florian.hopfner (at) inway.de

Reply:
You can also try this with the pipeline I provided above. If you change the export-Solution.yml template to use the parameter in the condition instead of the if statement, you will see that the step with such a condition will fail with an error.

Reply:
@FH-Inway oddly enough I just ran a test of this with your test harness and it worked in my case using conditions. So, I'm a little confused as to why you are seeing something different. Here's what I did

1. In the test pipeline I commented out the BranchToCreate parameter being passed to the template

![image](https://user-images.githubusercontent.com/42348035/122610082-0e8b4700-d04d-11eb-90af-2d966d4c249f.png)

2. I updated the template to use a condition

condition: and(succeeded(), ne('${{parameters.BranchToCreate}}', ''))

![image](https://user-images.githubusercontent.com/42348035/122610199-39759b00-d04d-11eb-8861-b0e4c03744ff.png)

3, The result was the step was skipped.

![image](https://user-images.githubusercontent.com/42348035/122609931-cbc96f00-d04c-11eb-9fcc-2b95496a6bbb.png)

Can you try the same? As an aside we are using parameters in conditions elsewhere in our pipelines without errors.

Reply:
@mikefactorial By commenting out the parameter in step 1, the default value of the parameter in the template is used, which is the empty string. This can be evalutated in the condition, because now no value is passed to the parameter from the pipeline.
Try uncommenting the parameter and then, when running the pipeline, provide a value. You will notice that instead of the provided value, in the condition the value '$(FeatureBranch)' will be used. Since there is no such variable (at least not at the time the yaml is compiled and the condition is evaluated, the condition expression will be 
`condition: and(succeeded(), ne('$(BranchToCreate)', ''))`
which will cause an error.

If the template is used by the `export-solution-to-git.yml` pipeline, the error would not occur, because the variable is not declared by the yaml of the pipeline, but outside of the pipeline. In this case, the variable exists at the time the condition is evaluated and the value of the variable is used. However, it would not be the value that was passed to the parameter, but instead, the original value of the variable.